From b91bda8506eda850bf444f160f06857a6927859f Mon Sep 17 00:00:00 2001
From: Brooks Keough <bkeough@nvidia.com>
Date: Mon, 13 Jan 2025 21:12:36 -0800
Subject: [PATCH] Added debug prints to capture invalid temps

This change adds prints for temperature sensors if the value is too
large (negative or positive). This is to assist in debugging the invalid
temp issue for virtual eeprom sensors.

Signed-off-by: Brooks Keough <bkeough@nvidia.com>
---
 .../lib/smbus_telemetry_update.cpp            | 19 +++++++++++++++++++
 1 file changed, 19 insertions(+)

diff --git a/smbus-telemetry-target/lib/smbus_telemetry_update.cpp b/smbus-telemetry-target/lib/smbus_telemetry_update.cpp
index d1111ec..7a05ad2 100644
--- a/smbus-telemetry-target/lib/smbus_telemetry_update.cpp
+++ b/smbus-telemetry-target/lib/smbus_telemetry_update.cpp
@@ -208,6 +208,25 @@ int smbusSlaveUpdate(std::string dbusObjPath, std::string iface,
             std::min((uint8_t)sensorDataMap[key].getOffsetDataLength(),
                      // commit min of actial bytes or mapped bytes
                      static_cast<uint8_t>(valBytes.size())));
+
+        const uint8_t* rawData =
+            reinterpret_cast<const uint8_t*>(valBytes.data());
+        if ((dbusObjPath.find("/HGX") !=
+             std::string::npos) && (valBytes.size() == 4) && (rawData[3] != 0))
+        {
+            lg2::error(
+                "Invalid sensor reading: {DBUSOBJECT}, {IFACE}, {PROPNAME}, {D1} {D2} {D3} {D4}",
+                "DBUSOBJECT", dbusObjPath.c_str(), "IFACE", iface.c_str(), "PROPNAME", propName.c_str(), "D1", rawData[0], "D2", rawData[1], "D3",
+                rawData[2], "D4", rawData[3]);
+        }
+        else if ((dbusObjPath.find("/HGX") !=
+             std::string::npos) && (valBytes.size() == 8) && (rawData[7] != 0)){
+            lg2::error(
+                "Invalid sensor reading: {DBUSOBJECT}, {IFACE}, {PROPNAME}, {D1} {D2} {D3} {D4} {D5} {D6} {D7} {D8}",
+                "DBUSOBJECT", dbusObjPath.c_str(), "IFACE", iface.c_str(), "PROPNAME", propName.c_str(), "D1", rawData[0], "D2", rawData[1], "D3",
+                rawData[2], "D4", rawData[3], "D5", rawData[4], "D6", rawData[5], "D7",
+                rawData[6], "D8", rawData[7]);
+        }
     }
     catch (const std::exception& e)
     {
-- 
2.25.1

